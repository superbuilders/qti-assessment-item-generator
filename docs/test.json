{
	"identifier": "number-line-distance-0-4",
	"title": "Identify 0.4 on a Number Line",
	"responseDeclarations": [
		{
			"identifier": "RESPONSE",
			"cardinality": "single",
			"baseType": "identifier",
			"correct": "C"
		}
	],
	"body": [
		{
			"type": "interactionRef",
			"interactionId": "choice_interaction"
		}
	],
	"interactions": {
		"choice_interaction": {
			"type": "choiceInteraction",
			"responseIdentifier": "RESPONSE",
			"prompt": [
				{
					"type": "text",
					"content": "Which number line shows a point that represents a distance of "
				},
				{
					"type": "math",
					"mathml": "<mn>0.4</mn>"
				},
				{
					"type": "text",
					"content": " units from zero? Select one answer."
				}
			],
			"choices": [
				{
					"identifier": "A",
					"content": [
						{
							"type": "widgetRef",
							"widgetId": "repl__choice_interaction__choice__a__visual__0",
							"widgetType": "numberLineWithAction"
						}
					]
				},
				{
					"identifier": "B",
					"content": [
						{
							"type": "widgetRef",
							"widgetId": "repl__choice_interaction__choice__b__visual__0",
							"widgetType": "numberLineWithAction"
						}
					]
				},
				{
					"identifier": "C",
					"content": [
						{
							"type": "widgetRef",
							"widgetId": "repl__choice_interaction__choice__c__visual__0",
							"widgetType": "numberLineWithAction"
						}
					]
				},
				{
					"identifier": "D",
					"content": [
						{
							"type": "widgetRef",
							"widgetId": "repl__choice_interaction__choice__d__visual__0",
							"widgetType": "numberLineWithAction"
						}
					]
				}
			],
			"shuffle": true,
			"minChoices": 1,
			"maxChoices": 1
		}
	},
	"widgets": {
		"repl__choice_interaction__choice__a__visual__0": {
			"type": "numberLineWithAction",
			"width": 338,
			"height": 320,
			"orientation": "horizontal",
			"min": 0,
			"max": 1,
			"tickInterval": {
				"type": "whole",
				"interval": 1
			},
			"secondaryTickInterval": {
				"type": "fraction",
				"numerator": 1,
				"denominator": 10
			},
			"startValue": 0,
			"actions": [
				{
					"delta": {
						"type": "fraction",
						"numerator": 1,
						"denominator": 10,
						"sign": "+"
					}
				},
				{
					"delta": {
						"type": "fraction",
						"numerator": 1,
						"denominator": 10,
						"sign": "+"
					}
				},
				{
					"delta": {
						"type": "fraction",
						"numerator": 1,
						"denominator": 10,
						"sign": "+"
					}
				}
			]
		},
		"repl__choice_interaction__choice__b__visual__0": {
			"type": "numberLineWithAction",
			"width": 338,
			"height": 320,
			"orientation": "horizontal",
			"min": 0,
			"max": 1,
			"tickInterval": {
				"type": "whole",
				"interval": 1
			},
			"secondaryTickInterval": {
				"type": "fraction",
				"numerator": 1,
				"denominator": 10
			},
			"startValue": 0,
			"actions": [
				{
					"delta": {
						"type": "fraction",
						"numerator": 1,
						"denominator": 10,
						"sign": "+"
					}
				},
				{
					"delta": {
						"type": "fraction",
						"numerator": 1,
						"denominator": 10,
						"sign": "+"
					}
				}
			]
		},
		"repl__choice_interaction__choice__c__visual__0": {
			"type": "numberLineWithAction",
			"width": 338,
			"height": 320,
			"orientation": "horizontal",
			"min": 0,
			"max": 1,
			"tickInterval": {
				"type": "whole",
				"interval": 1
			},
			"secondaryTickInterval": {
				"type": "fraction",
				"numerator": 1,
				"denominator": 10
			},
			"startValue": 0,
			"actions": [
				{
					"delta": {
						"type": "fraction",
						"numerator": 1,
						"denominator": 10,
						"sign": "+"
					}
				},
				{
					"delta": {
						"type": "fraction",
						"numerator": 1,
						"denominator": 10,
						"sign": "+"
					}
				},
				{
					"delta": {
						"type": "fraction",
						"numerator": 1,
						"denominator": 10,
						"sign": "+"
					}
				},
				{
					"delta": {
						"type": "fraction",
						"numerator": 1,
						"denominator": 10,
						"sign": "+"
					}
				}
			]
		},
		"repl__choice_interaction__choice__d__visual__0": {
			"type": "numberLineWithAction",
			"width": 338,
			"height": 320,
			"orientation": "horizontal",
			"min": 0,
			"max": 1,
			"tickInterval": {
				"type": "whole",
				"interval": 1
			},
			"secondaryTickInterval": {
				"type": "fraction",
				"numerator": 1,
				"denominator": 10
			},
			"startValue": 0,
			"actions": [
				{
					"delta": {
						"type": "fraction",
						"numerator": 1,
						"denominator": 10,
						"sign": "+"
					}
				},
				{
					"delta": {
						"type": "fraction",
						"numerator": 1,
						"denominator": 10,
						"sign": "+"
					}
				},
				{
					"delta": {
						"type": "fraction",
						"numerator": 1,
						"denominator": 10,
						"sign": "+"
					}
				},
				{
					"delta": {
						"type": "fraction",
						"numerator": 1,
						"denominator": 10,
						"sign": "+"
					}
				},
				{
					"delta": {
						"type": "fraction",
						"numerator": 1,
						"denominator": 10,
						"sign": "+"
					}
				},
				{
					"delta": {
						"type": "fraction",
						"numerator": 1,
						"denominator": 10,
						"sign": "+"
					}
				}
			]
		}
	},
	"feedbackPlan": {
		"mode": "combo",
		"dimensions": [
			{
				"responseIdentifier": "RESPONSE",
				"kind": "enumerated",
				"keys": ["A", "B", "C", "D"]
			}
		],
		"combinations": [
			{
				"id": "FB__RESPONSE_A",
				"path": [
					{
						"responseIdentifier": "RESPONSE",
						"key": "A"
					}
				]
			},
			{
				"id": "FB__RESPONSE_B",
				"path": [
					{
						"responseIdentifier": "RESPONSE",
						"key": "B"
					}
				]
			},
			{
				"id": "FB__RESPONSE_C",
				"path": [
					{
						"responseIdentifier": "RESPONSE",
						"key": "C"
					}
				]
			},
			{
				"id": "FB__RESPONSE_D",
				"path": [
					{
						"responseIdentifier": "RESPONSE",
						"key": "D"
					}
				]
			}
		]
	},
	"feedback": {
		"FEEDBACK__OVERALL": {
			"RESPONSE": {
				"A": {
					"content": {
						"preamble": {
							"correctness": "incorrect",
							"summary": [
								{
									"type": "text",
									"content": "It looks like the selected number line does not place the point at "
								},
								{
									"type": "math",
									"mathml": "<mn>0.4</mn>"
								},
								{
									"type": "text",
									"content": " units from "
								},
								{
									"type": "math",
									"mathml": "<mn>0</mn>"
								},
								{
									"type": "text",
									"content": ", which is "
								},
								{
									"type": "math",
									"mathml": "<mfrac><mn>4</mn><mn>10</mn></mfrac>"
								},
								{
									"type": "text",
									"content": " of the way from "
								},
								{
									"type": "math",
									"mathml": "<mn>0</mn>"
								},
								{
									"type": "text",
									"content": " to "
								},
								{
									"type": "math",
									"mathml": "<mn>1</mn>"
								},
								{
									"type": "text",
									"content": "."
								}
							]
						},
						"steps": [
							{
								"type": "step",
								"title": [
									{
										"type": "text",
										"content": "Interpret "
									},
									{
										"type": "math",
										"mathml": "<mn>0.4</mn>"
									},
									{
										"type": "text",
										"content": " as four tenths."
									}
								],
								"content": [
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "Recall that each tenth is "
											},
											{
												"type": "math",
												"mathml": "<mn>0.1</mn>"
											},
											{
												"type": "text",
												"content": " unit."
											}
										]
									},
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "So "
											},
											{
												"type": "math",
												"mathml": "<mn>0.4</mn>"
											},
											{
												"type": "text",
												"content": " equals "
											},
											{
												"type": "math",
												"mathml": "<mfrac><mn>4</mn><mn>10</mn></mfrac>"
											},
											{
												"type": "text",
												"content": ", meaning four jumps of size "
											},
											{
												"type": "math",
												"mathml": "<mn>0.1</mn>"
											},
											{
												"type": "text",
												"content": " from "
											},
											{
												"type": "math",
												"mathml": "<mn>0</mn>"
											},
											{
												"type": "text",
												"content": "."
											}
										]
									},
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "It lies between "
											},
											{
												"type": "math",
												"mathml": "<mn>0.3</mn>"
											},
											{
												"type": "text",
												"content": " and "
											},
											{
												"type": "math",
												"mathml": "<mn>0.5</mn>"
											},
											{
												"type": "text",
												"content": ", and "
											},
											{
												"type": "math",
												"mathml": "<mn>0.4</mn><mo>&lt;</mo><mn>0.5</mn>"
											},
											{
												"type": "text",
												"content": ", so it is a little less than halfway from "
											},
											{
												"type": "math",
												"mathml": "<mn>0</mn>"
											},
											{
												"type": "text",
												"content": " to "
											},
											{
												"type": "math",
												"mathml": "<mn>1</mn>"
											},
											{
												"type": "text",
												"content": "."
											}
										]
									}
								]
							},
							{
								"type": "step",
								"title": [
									{
										"type": "text",
										"content": "Use the scale of the number line."
									}
								],
								"content": [
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "Look for a number line labeled from "
											},
											{
												"type": "math",
												"mathml": "<mn>0</mn>"
											},
											{
												"type": "text",
												"content": " to "
											},
											{
												"type": "math",
												"mathml": "<mn>1</mn>"
											},
											{
												"type": "text",
												"content": " that is divided into ten equal parts; each small tick then represents "
											},
											{
												"type": "math",
												"mathml": "<mn>0.1</mn>"
											},
											{
												"type": "text",
												"content": "."
											}
										]
									},
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "Starting at "
											},
											{
												"type": "math",
												"mathml": "<mn>0</mn>"
											},
											{
												"type": "text",
												"content": ", count "
											},
											{
												"type": "math",
												"mathml": "<mn>4</mn>"
											},
											{
												"type": "text",
												"content": " equal ticks to the right; that location corresponds to "
											},
											{
												"type": "math",
												"mathml": "<mn>0.4</mn>"
											},
											{
												"type": "text",
												"content": "."
											}
										]
									}
								]
							},
							{
								"type": "step",
								"title": [
									{
										"type": "text",
										"content": "Self-check before selecting."
									}
								],
								"content": [
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "Does the segment from "
											},
											{
												"type": "math",
												"mathml": "<mn>0</mn>"
											},
											{
												"type": "text",
												"content": " to "
											},
											{
												"type": "math",
												"mathml": "<mn>1</mn>"
											},
											{
												"type": "text",
												"content": " have ten equal parts (tenths)?"
											}
										]
									},
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "Is the marked point exactly "
											},
											{
												"type": "math",
												"mathml": "<mn>4</mn>"
											},
											{
												"type": "text",
												"content": " of those equal parts to the right of "
											},
											{
												"type": "math",
												"mathml": "<mn>0</mn>"
											},
											{
												"type": "text",
												"content": ", and to the left of "
											},
											{
												"type": "math",
												"mathml": "<mn>0.5</mn>"
											},
											{
												"type": "text",
												"content": " since "
											},
											{
												"type": "math",
												"mathml": "<mn>0.4</mn><mo>&lt;</mo><mn>0.5</mn>"
											},
											{
												"type": "text",
												"content": "?"
											}
										]
									}
								]
							}
						]
					}
				},
				"B": {
					"content": {
						"preamble": {
							"correctness": "incorrect",
							"summary": [
								{
									"type": "text",
									"content": "The point should be exactly "
								},
								{
									"type": "math",
									"mathml": "<mn>0.4</mn>"
								},
								{
									"type": "text",
									"content": " units from "
								},
								{
									"type": "math",
									"mathml": "<mn>0</mn>"
								},
								{
									"type": "text",
									"content": ", which is four tenths; the number line you selected does not place the point that distance from zero."
								}
							]
						},
						"steps": [
							{
								"type": "step",
								"title": [
									{
										"type": "text",
										"content": "Determine the size of one tick mark"
									}
								],
								"content": [
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "Look between "
											},
											{
												"type": "math",
												"mathml": "<mn>0</mn>"
											},
											{
												"type": "text",
												"content": " and "
											},
											{
												"type": "math",
												"mathml": "<mn>1</mn>"
											},
											{
												"type": "text",
												"content": ". Count the equal spaces. If there are "
											},
											{
												"type": "math",
												"mathml": "<mn>10</mn>"
											},
											{
												"type": "text",
												"content": ", then each space is "
											},
											{
												"type": "math",
												"mathml": "<mn>0.1</mn>"
											},
											{
												"type": "text",
												"content": " (one tenth)."
											}
										]
									},
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "Four tick marks from zero represent "
											},
											{
												"type": "math",
												"mathml": "<mn>4</mn>"
											},
											{
												"type": "text",
												"content": " tenths: "
											},
											{
												"type": "math",
												"mathml": "<mfrac><mn>4</mn><mn>10</mn></mfrac>"
											},
											{
												"type": "text",
												"content": " which equals "
											},
											{
												"type": "math",
												"mathml": "<mn>0.4</mn>"
											},
											{
												"type": "text",
												"content": "."
											}
										]
									}
								]
							},
							{
								"type": "step",
								"title": [
									{
										"type": "text",
										"content": "Use distance from zero"
									}
								],
								"content": [
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "Distance ignores direction: points at "
											},
											{
												"type": "math",
												"mathml": "<mo>+</mo><mn>0.4</mn>"
											},
											{
												"type": "text",
												"content": " and "
											},
											{
												"type": "math",
												"mathml": "<mo>−</mo><mn>0.4</mn>"
											},
											{
												"type": "text",
												"content": " are both a distance of "
											},
											{
												"type": "math",
												"mathml": "<mn>0.4</mn>"
											},
											{
												"type": "text",
												"content": " from "
											},
											{
												"type": "math",
												"mathml": "<mn>0</mn>"
											},
											{
												"type": "text",
												"content": " because "
											},
											{
												"type": "math",
												"mathml": "<mo>|</mo><mi>x</mi><mo>|</mo>"
											},
											{
												"type": "text",
												"content": " measures how far "
											},
											{
												"type": "math",
												"mathml": "<mi>x</mi>"
											},
											{
												"type": "text",
												"content": " is from zero."
											}
										]
									},
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "Starting at "
											},
											{
												"type": "math",
												"mathml": "<mn>0</mn>"
											},
											{
												"type": "text",
												"content": ", count "
											},
											{
												"type": "math",
												"mathml": "<mn>4</mn>"
											},
											{
												"type": "text",
												"content": " tick marks of size "
											},
											{
												"type": "math",
												"mathml": "<mn>0.1</mn>"
											},
											{
												"type": "text",
												"content": " to locate the point. It should land exactly on the 4th tick, not past it."
											}
										]
									}
								]
							},
							{
								"type": "step",
								"title": [
									{
										"type": "text",
										"content": "Self-check and revise your choice"
									}
								],
								"content": [
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "Ask yourself:"
											}
										]
									},
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "• Do you see ten equal spaces between "
											},
											{
												"type": "math",
												"mathml": "<mn>0</mn>"
											},
											{
												"type": "text",
												"content": " and "
											},
											{
												"type": "math",
												"mathml": "<mn>1</mn>"
											},
											{
												"type": "text",
												"content": "?"
											}
										]
									},
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "• Is the point exactly the 4th tick from "
											},
											{
												"type": "math",
												"mathml": "<mn>0</mn>"
											},
											{
												"type": "text",
												"content": " (to the right or to the left), not at "
											},
											{
												"type": "math",
												"mathml": "<mn>0.04</mn>"
											},
											{
												"type": "text",
												"content": " or at "
											},
											{
												"type": "math",
												"mathml": "<mn>4</mn>"
											},
											{
												"type": "text",
												"content": "?"
											}
										]
									},
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "Use these checks to select the number line that shows a point exactly "
											},
											{
												"type": "math",
												"mathml": "<mn>0.4</mn>"
											},
											{
												"type": "text",
												"content": " units from zero."
											}
										]
									}
								]
							}
						]
					}
				},
				"C": {
					"content": {
						"preamble": {
							"correctness": "correct",
							"summary": [
								{
									"type": "text",
									"content": " "
								},
								{
									"type": "math",
									"mathml": "<mn>0.4</mn>"
								},
								{
									"type": "text",
									"content": " represents four tenths of a unit from zero. On a number line from "
								},
								{
									"type": "math",
									"mathml": "<mn>0</mn>"
								},
								{
									"type": "text",
									"content": " to "
								},
								{
									"type": "math",
									"mathml": "<mn>1</mn>"
								},
								{
									"type": "text",
									"content": " split into ten equal parts, this is the fourth tick to the right of zero."
								}
							]
						},
						"steps": [
							{
								"type": "step",
								"title": [
									{
										"type": "text",
										"content": "Interpret the decimal as tenths"
									}
								],
								"content": [
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "Recall that "
											},
											{
												"type": "math",
												"mathml": "<mn>0.4</mn><mo>=</mo><mfrac><mn>4</mn><mn>10</mn></mfrac>"
											},
											{
												"type": "text",
												"content": ", which means four tenths of one whole unit."
											}
										]
									}
								]
							},
							{
								"type": "step",
								"title": [
									{
										"type": "text",
										"content": "Locate the point on a 0–1 number line"
									}
								],
								"content": [
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "If the interval from "
											},
											{
												"type": "math",
												"mathml": "<mn>0</mn>"
											},
											{
												"type": "text",
												"content": " to "
											},
											{
												"type": "math",
												"mathml": "<mn>1</mn>"
											},
											{
												"type": "text",
												"content": " is divided into ten equal segments, each tick represents "
											},
											{
												"type": "math",
												"mathml": "<mn>0.1</mn>"
											},
											{
												"type": "text",
												"content": "."
											}
										]
									},
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "Starting at zero, move "
											},
											{
												"type": "math",
												"mathml": "<mn>4</mn>"
											},
											{
												"type": "text",
												"content": " ticks to the right to land at "
											},
											{
												"type": "math",
												"mathml": "<mn>0.4</mn>"
											},
											{
												"type": "text",
												"content": "."
											}
										]
									}
								]
							},
							{
								"type": "step",
								"title": [
									{
										"type": "text",
										"content": "Connect to your selection"
									}
								],
								"content": [
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "The diagram you chose places the point exactly four tenths of a unit from zero, matching "
											},
											{
												"type": "math",
												"mathml": "<mn>0.4</mn>"
											},
											{
												"type": "text",
												"content": "."
											}
										]
									}
								]
							}
						]
					}
				},
				"D": {
					"content": {
						"preamble": {
							"correctness": "incorrect",
							"summary": [
								{
									"type": "text",
									"content": "The diagram you chose does not show a point exactly four tenths of a unit from zero."
								},
								{
									"type": "text",
									"content": " On a number line from "
								},
								{
									"type": "math",
									"mathml": "<mn>0</mn>"
								},
								{
									"type": "text",
									"content": " to "
								},
								{
									"type": "math",
									"mathml": "<mn>1</mn>"
								},
								{
									"type": "text",
									"content": " split into ten equal parts, each tick represents "
								},
								{
									"type": "math",
									"mathml": "<mn>0.1</mn>"
								},
								{
									"type": "text",
									"content": ", so "
								},
								{
									"type": "math",
									"mathml": "<mn>0.4</mn>"
								},
								{
									"type": "text",
									"content": " is at the fourth tick to the right of "
								},
								{
									"type": "math",
									"mathml": "<mn>0</mn>"
								},
								{
									"type": "text",
									"content": "."
								}
							]
						},
						"steps": [
							{
								"type": "step",
								"title": [
									{
										"type": "text",
										"content": "Interpret the distance"
									}
								],
								"content": [
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "“Distance from zero” means how far the point is from "
											},
											{
												"type": "math",
												"mathml": "<mn>0</mn>"
											},
											{
												"type": "text",
												"content": ", not which direction."
											}
										]
									},
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "Think of "
											},
											{
												"type": "math",
												"mathml": "<mn>0.4</mn>"
											},
											{
												"type": "text",
												"content": " as four tenths."
											}
										]
									}
								]
							},
							{
								"type": "step",
								"title": [
									{
										"type": "text",
										"content": "Use tenths on the 0–1 number line"
									}
								],
								"content": [
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "Check that the segment from "
											},
											{
												"type": "math",
												"mathml": "<mn>0</mn>"
											},
											{
												"type": "text",
												"content": " to "
											},
											{
												"type": "math",
												"mathml": "<mn>1</mn>"
											},
											{
												"type": "text",
												"content": " is divided into ten equal spaces; each space is "
											},
											{
												"type": "math",
												"mathml": "<mn>0.1</mn>"
											},
											{
												"type": "text",
												"content": "."
											}
										]
									},
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "Starting at "
											},
											{
												"type": "math",
												"mathml": "<mn>0</mn>"
											},
											{
												"type": "text",
												"content": ", count four equal intervals to the right. That location corresponds to "
											},
											{
												"type": "math",
												"mathml": "<mn>0.4</mn>"
											},
											{
												"type": "text",
												"content": "."
											}
										]
									},
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "Avoid common slips like stopping at the fifth tick ("
											},
											{
												"type": "math",
												"mathml": "<mn>0.5</mn>"
											},
											{
												"type": "text",
												"content": ") or using uneven partitions."
											}
										]
									}
								]
							},
							{
								"type": "step",
								"title": [
									{
										"type": "text",
										"content": "Self-check and revise your choice"
									}
								],
								"content": [
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "Ask yourself:"
											}
										]
									},
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "• Does the diagram have exactly ten equal spaces between "
											},
											{
												"type": "math",
												"mathml": "<mn>0</mn>"
											},
											{
												"type": "text",
												"content": " and "
											},
											{
												"type": "math",
												"mathml": "<mn>1</mn>"
											},
											{
												"type": "text",
												"content": "?"
											}
										]
									},
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "• Is the point exactly four tick marks to the right of "
											},
											{
												"type": "math",
												"mathml": "<mn>0</mn>"
											},
											{
												"type": "text",
												"content": ", representing "
											},
											{
												"type": "math",
												"mathml": "<mn>0.4</mn>"
											},
											{
												"type": "text",
												"content": "?"
											}
										]
									},
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "Use these checks to select the diagram that matches this distance."
											}
										]
									}
								]
							}
						]
					}
				}
			}
		}
	}
}
