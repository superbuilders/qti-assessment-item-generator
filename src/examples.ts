import type { AssessmentItemInput } from "@/core/item"
import type { allWidgetsCollection } from "@/widgets/collections/all"
import type { WidgetTypeTupleFrom } from "@/widgets/collections/types"

export const allExamples: AssessmentItemInput<WidgetTypeTupleFrom<typeof allWidgetsCollection>>[] = []
