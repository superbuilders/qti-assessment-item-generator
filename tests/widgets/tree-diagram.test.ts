import { describe, expect, test } from "bun:test"
import * as errors from "@superbuilders/errors"
import * as logger from "@superbuilders/slog"
import { treeDiagramExamples } from "../../examples/tree-diagram"
import type { WidgetInput } from "../../src/widgets/registry"
import { generateWidgetForTest } from "../helpers/generateWidgetForTest"

describe("Widget: tree-diagram", () => {
	const examples: WidgetInput[] = treeDiagramExamples

	examples.forEach((props, index) => {
		test(`should produce consistent output for example #${index + 1}`, async () => {
			const result = await errors.try(generateWidgetForTest(props))
			if (result.error) {
				logger.error("widget generation failed", { error: result.error, index })
				throw result.error
			}
			expect(result.data).toMatchSnapshot()
		})
	})
})
