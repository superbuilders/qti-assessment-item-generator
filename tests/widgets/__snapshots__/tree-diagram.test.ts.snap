// Bun Snapshot v1, https://bun.sh/docs/test/snapshots

exports[`Widget: tree-diagram should produce consistent output for example #1 1`] = `"<svg width="480" height="353" viewBox="-20 -29 480 353" xmlns="http://www.w3.org/2000/svg" font-family="sans-serif"><line x1="220" y1="12" x2="120" y2="74" stroke="#000000" stroke-width="2"/><line x1="220" y1="12" x2="320" y2="74" stroke="#000000" stroke-width="2"/><line x1="120" y1="74" x2="70" y2="174" stroke="#000000" stroke-width="2"/><line x1="120" y1="74" x2="170" y2="174" stroke="#000000" stroke-width="2"/><line x1="320" y1="74" x2="270" y2="174" stroke="#000000" stroke-width="2"/><line x1="320" y1="74" x2="370" y2="174" stroke="#000000" stroke-width="2"/><line x1="70" y1="174" x2="45" y2="274" stroke="#000000" stroke-width="2"/><line x1="70" y1="174" x2="95" y2="274" stroke="#000000" stroke-width="2"/><line x1="170" y1="174" x2="145" y2="274" stroke="#000000" stroke-width="2"/><line x1="170" y1="174" x2="195" y2="274" stroke="#000000" stroke-width="2"/><line x1="270" y1="174" x2="245" y2="274" stroke="#000000" stroke-width="2"/><line x1="270" y1="174" x2="295" y2="274" stroke="#000000" stroke-width="2"/><line x1="370" y1="174" x2="345" y2="274" stroke="#000000" stroke-width="2"/><line x1="370" y1="174" x2="395" y2="274" stroke="#000000" stroke-width="2"/><circle cx="220" cy="12" r="20" fill="#ffffff" stroke="#000000" stroke-width="2"/><text x="220" y="12" font-size="14" text-anchor="middle" fill="#000000" dominant-baseline="middle"><tspan x="220" dy="0">Start</tspan></text><circle cx="120" cy="74" r="20" fill="#ffffff" stroke="#000000" stroke-width="2"/><text x="120" y="74" font-size="14" text-anchor="middle" fill="#000000" dominant-baseline="middle"><tspan x="120" dy="0">H</tspan></text><circle cx="320" cy="74" r="20" fill="#ffffff" stroke="#000000" stroke-width="2"/><text x="320" y="74" font-size="14" text-anchor="middle" fill="#000000" dominant-baseline="middle"><tspan x="320" dy="0">T</tspan></text><circle cx="70" cy="174" r="20" fill="#ffffff" stroke="#000000" stroke-width="2"/><text x="70" y="174" font-size="14" text-anchor="middle" fill="#000000" dominant-baseline="middle"><tspan x="70" dy="0">H</tspan></text><circle cx="170" cy="174" r="20" fill="#ffffff" stroke="#000000" stroke-width="2"/><text x="170" y="174" font-size="14" text-anchor="middle" fill="#000000" dominant-baseline="middle"><tspan x="170" dy="0">T</tspan></text><circle cx="270" cy="174" r="20" fill="#ffffff" stroke="#000000" stroke-width="2"/><text x="270" y="174" font-size="14" text-anchor="middle" fill="#000000" dominant-baseline="middle"><tspan x="270" dy="0">H</tspan></text><circle cx="370" cy="174" r="20" fill="#ffffff" stroke="#000000" stroke-width="2"/><text x="370" y="174" font-size="14" text-anchor="middle" fill="#000000" dominant-baseline="middle"><tspan x="370" dy="0">T</tspan></text><circle cx="45" cy="274" r="20" fill="#ffffff" stroke="#000000" stroke-width="2"/><text x="45" y="274" font-size="14" text-anchor="middle" fill="#000000" dominant-baseline="middle"><tspan x="45" dy="0">H</tspan></text><circle cx="95" cy="274" r="20" fill="#ffffff" stroke="#000000" stroke-width="2"/><text x="95" y="274" font-size="14" text-anchor="middle" fill="#000000" dominant-baseline="middle"><tspan x="95" dy="0">T</tspan></text><circle cx="145" cy="274" r="20" fill="#ffffff" stroke="#000000" stroke-width="2"/><text x="145" y="274" font-size="14" text-anchor="middle" fill="#000000" dominant-baseline="middle"><tspan x="145" dy="0">H</tspan></text><circle cx="195" cy="274" r="20" fill="#ffffff" stroke="#000000" stroke-width="2"/><text x="195" y="274" font-size="14" text-anchor="middle" fill="#000000" dominant-baseline="middle"><tspan x="195" dy="0">T</tspan></text><circle cx="245" cy="274" r="20" fill="#ffffff" stroke="#000000" stroke-width="2"/><text x="245" y="274" font-size="14" text-anchor="middle" fill="#000000" dominant-baseline="middle"><tspan x="245" dy="0">H</tspan></text><circle cx="295" cy="274" r="20" fill="#ffffff" stroke="#000000" stroke-width="2"/><text x="295" y="274" font-size="14" text-anchor="middle" fill="#000000" dominant-baseline="middle"><tspan x="295" dy="0">T</tspan></text><circle cx="345" cy="274" r="20" fill="#ffffff" stroke="#000000" stroke-width="2"/><text x="345" y="274" font-size="14" text-anchor="middle" fill="#000000" dominant-baseline="middle"><tspan x="345" dy="0">H</tspan></text><circle cx="395" cy="274" r="20" fill="#ffffff" stroke="#000000" stroke-width="2"/><text x="395" y="274" font-size="14" text-anchor="middle" fill="#000000" dominant-baseline="middle"><tspan x="395" dy="0">T</tspan></text></svg>"`;

exports[`Widget: tree-diagram should produce consistent output for example #2 1`] = `"<svg width="480" height="213" viewBox="-20 -49 480 213" xmlns="http://www.w3.org/2000/svg" font-family="sans-serif"><line x1="220" y1="12" x2="70" y2="94" stroke="#000000" stroke-width="2"/><line x1="220" y1="12" x2="170" y2="94" stroke="#000000" stroke-width="2"/><line x1="220" y1="12" x2="270" y2="94" stroke="#000000" stroke-width="2"/><line x1="220" y1="12" x2="370" y2="94" stroke="#000000" stroke-width="2"/><circle cx="220" cy="12" r="40" fill="#ffffff" stroke="#000000" stroke-width="2"/><circle cx="70" cy="94" r="40" fill="#ffffff" stroke="#000000" stroke-width="2"/><text x="70" y="94" font-size="12" text-anchor="middle" fill="#000000" dominant-baseline="middle"><tspan x="70" dy="0">3 Lefts</tspan></text><circle cx="170" cy="94" r="40" fill="#ffffff" stroke="#000000" stroke-width="2"/><text x="170" y="94" font-size="12" text-anchor="middle" fill="#000000" dominant-baseline="middle"><tspan x="170" dy="0">2 Lefts 1</tspan><tspan x="170" dy="1.2em">Right</tspan></text><circle cx="270" cy="94" r="40" fill="#ffffff" stroke="#000000" stroke-width="2"/><text x="270" y="94" font-size="12" text-anchor="middle" fill="#000000" dominant-baseline="middle"><tspan x="270" dy="0">1 Left 2</tspan><tspan x="270" dy="1.2em">Rights</tspan></text><circle cx="370" cy="94" r="40" fill="#ffffff" stroke="#000000" stroke-width="2"/><text x="370" y="94" font-size="12" text-anchor="middle" fill="#000000" dominant-baseline="middle"><tspan x="370" dy="0">3 Rights</tspan></text></svg>"`;
